packages/@aws-cdk/aws-ecs/lib/load-balanced-fargate-service.ts

```diff
@@ -76,6 +77,22 @@ export interface LoadBalancedFargateServiceProps {
    * @default false
    */
   publicTasks?: boolean;
+
+  /**
+   * Domain name for the service, e.g. api.example.com
+   */
+  domainName?: string;
+
+  /**
+   * Route53 hosted zone name for the domain, e.g. "example.com."
+   */
+  domainZone?: string;
```

Comment:
Use [HostedZoneRef](https://awslabs.github.io/aws-cdk/refs/_aws-cdk_aws-route53.html#hostedzoneref) instead of a string

Reply:
How would that work for hosted zones created outside of CFN?  That's pretty common since record sets can be created in CFN against the hosted zone name

Reply:
See #1131

Reply:
We do have a HostedZoneContextProvider these days, I think it allows looking up a R53 zone by name

Reply:
https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-route53/lib/hosted-zone-provider.ts

Unfortunately no README documentation :(

Reply:
Union types are poorly supported in some languages (e.g. Java), so we try to avoid them in L2 APIs

Reply:
Done, using HostedZoneRef